0001-typedb/02-dev/02-schema.md

```diff
@@ -0,0 +1,948 @@
+```
+
+#### Examples
+
+##### Basic
+
+The following `define` query creates a new entity type, `object`, by subtyping the `entity` base type.
+
+<!-- test-ignore -->
+```typeql
+define object sub entity;
+```
+
+##### Abstract
+
+Optionally, entity types can be defined as `abstract`. An abstract entity type can’t be instantiated and must be 
+subtyped in order to create entities. It exists only so other entity types can inherit the attribute types it owns 
+and the roles it plays.
+
+<!-- test-ignore -->
+```typeql
+define object sub entity, abstract;
+```
+
+##### Owns an attribute
+
+To define a new entity type that owns one or more attribute types, use the `owns` keyword followed by the name of the 
+attribute type. The attribute types are appended to the entity type definition with commas. Note, attribute types must 
+be defined before or concurrently (in the same transaction) with its owner(s). You can add owners later but you can't 
+own non-existent attribute type.
+
+<!-- test-ignore -->
+```typeql
+define
+
+object-type sub attribute, value string;
+object sub entity, abstract, owns object-type;
+```
+
+###### Cardinality
+
+By default, an entity can have multiple attributes of the same type: zero, one, or many.
+
+By having multiple attributes of the same type we're effectively creating a 
+[multivalued attribute](04-write.md#multivalued-attributes) (i.e. having an attribute type instantiated with multiple 
+values).
+
+Use the `@key` keyword to limit the cardinality to exactly **one** and add **uniqueness** constraint. 
+See example in [Key attribute](#define-attribute-types) section.
+
+##### Plays a role
+
+To add roles that entities of a specific entity type can play, use the `plays` keyword.
+
+<!-- test-ignore -->
+```typeql
+define
+
+access sub relation, relates accessed-object;
+object sub entity, abstract, plays access:accessed-object;
+```
+
+##### Subtypes another entity
+
+<div class="note">
+[Note]
+All types that are subtyping `entity` base type directly or through other subtypes are called entity types. 
+Instances of these types are called entities. The same approach can be applied to attributes and relations.
+</div>
+
+An entity type can subtype another entity type by using the same `sub` keyword, but replacing the `entity` after it 
+with the name of another entity type to subtype.
+
+<!-- test-ignore -->
+```typeql
+define
+
+path sub attribute, value string;
+
+object sub entity, abstract, owns object-type, plays access:accessed-object;
+resource sub object, abstract;
+file sub resource, owns path;
+```
+
+In the above example, the `resource` and `file` entity types are subtypes of the `object`, which itself is a subtype 
+of the `entity` base type. They inherit the `object-type` attribute type ownership from it as well as its 
+`access:accessed-object` role. However, while the `resource` subtype is abstract, the `file` subtype is not. Hence, we 
+can create `file` entities, but not `resource` entities.
```

Comment:
This is fine as long as there is consistency between usages of 'instances' and 'entities' across the documentation.

Reply:
Yes. Entity is an instance of Entity type (being the base type or some of its subtypes). I'm usually use instances when I need to make sure the difference is crystal clear. But Shane doesn't like the word. And sometimes he deletes it from texts. I have tried to keep it anywhere it's important.